### ISSUE TITLE
[Question]: replication of model evaluation

### ISSUE THREAD

**Issue by cjing1** [flairNLP/flair#3233]
### Question

I tried to replicate the evaluation such as accuracy and F-1 score, etc. by loading the final-model.pt, predicting on test.csv data, and calculating using sklearn. But the results are all very different. Any intuition why that is?

**Comment by helpmefindaname** [comment]
Hi @cjing1 
My intuition is that there is a discrepancy between the flair evaluation code and your try to replicate it.

I suppose I could give you a better guess, if you share how you tried it, what the metrics are and what task you are evaluating.

**Comment by cjing1** [comment]
Thank you for your reply. 

It actually happened across all of the methods I tried including DocumentLSTMEmbeddings, StackedEmbeddings, TransformerDocumentEmbeddings, etc. The task is binary text classification. 

The following is the example code: 

word_embeddings = [WordEmbeddings('glove'),FlairEmbeddings('news-forward'),FlairEmbeddings('news-backward'),BertEmbeddings('distilbert-base-uncased')]

document_embeddings = DocumentLSTMEmbeddings(word_embeddings,hidden_size=512,reproject_words=True,reproject_words_dimension=256)

clf = TextClassifier(document_embeddings, label_dictionary = label_dict_csv, label_type = 'labels')
trainer = ModelTrainer(clf,corpus)
trainer.train('data/',learning_rate=5.0e-5,
                 mini_batch_size=4,
                 max_epochs=50)

After done, it produces a training log with model evaluation: 
- F-score (micro) 0.8285
- F-score (macro) 0.8197
- Accuracy 0.8285

However, when I load the final-model.pt and apply it on test.csv, then use sklearn.metrics

metrics.f1_score(labels, predict)
0.8117154811715481
metrics.accuracy_score(labels, predict)
0.8408488063660478

**Comment by helpmefindaname** [comment]
First of all, I want to point out that flair always does multiclass evaluation, so you need to use `micro_f1 = metrics.f1_score(..,, labels=["label a", "label b"], average="micro")` and `macro_f1 = metrics.f1_score(..,, labels=["label a", "label b"], average="macro")` for the f1 scores.
However as the accuracy is different, I suppose there is still another mistake in the replication. Can you share the code how you created `labels` and `predict`?

**Comment by cjing1** [comment]
Thank you for the clarification. The following are the code: 

from flair.data import Corpus
from flair.datasets import ClassificationCorpus,CSVClassificationCorpus

# this is the folder in which train, test and dev files reside
data_folder = 'data/'

# column format indicating which columns hold the text and label(s)
column_name_map = {1: 'text', 2: 'labels'}

# load corpus containing training, test and dev data and if CSV has a header, you can skip it
corpus: Corpus = CSVClassificationCorpus(data_folder, column_name_map=column_name_map,
                                         skip_header = True, delimiter=',', label_type='labels'
)

# Creating the Label Diction For CSV
label_dict_csv = corpus.make_label_dictionary(label_type = 'labels')

**Comment by helpmefindaname** [comment]
The code you posted creates neither `labels` nor `predict`, I cannot analyze your replication, if you don't want to share it.

Besides that, please use the [Markdown code blocks](https://www.markdownguide.org/extended-syntax/#fenced-code-blocks):
You can wrap your code around with quotes like this:
````
```python
# my python code here
```
````
That way the code won't be ill-formatted and is easier to read.

**Comment by cjing1** [comment]
> The code you posted creates neither `labels` nor `predict`, I cannot analyze your replication, if you don't want to share it.
> 
> Besides that, please use the [Markdown code blocks](https://www.markdownguide.org/extended-syntax/#fenced-code-blocks): You can wrap your code around with quotes like this:
> 
> ```
> ```python
> # my python code here
> ```
> ```
> 
> That way the code won't be ill-formatted and is easier to read.

Sorry, I misunderstand, the following are the codes: 

```python

labels = pd.Series(np.where(df1['labels'].values == 'Yes', 1, 0),
          df1.index)

predict = pd.Series(np.where(df1['predict_value'].values == 'Yes', 1, 0),
          df1.index)

```
